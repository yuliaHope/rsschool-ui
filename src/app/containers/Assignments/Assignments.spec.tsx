import * as React from 'react';
import { shallow } from 'enzyme';
import AssignmentsContainer from './index';
const { __stateGetter } = require('react-redux');

describe('Assignments container', () => {
    beforeEach(() => {
        __stateGetter.mockImplementation(() => ({}));
    });
    const props = {
        courseId: 'courseId',
        isAdmin: false,
        userProfile: 'userProfile',
        assignments: [],
        isLoading: false,
        onLoad: jest.fn(),
        updateAssignment: jest.fn(),
    };

    describe('Initial render', () => {
        it('renders properly', () => {
            const output = shallow(<AssignmentsContainer {...props} />).dive();
            expect(output).toMatchSnapshot();
        });
    });

    describe('onLoad called from CDM', () => {
        it('dispatches onLoad', () => {
            expect(props.onLoad).toHaveBeenCalledTimes(1);
        });
    });

    describe('Assignments preloader', () => {
        const nextProps = {
            ...props,
            isLoading: true,
        };

        it('renders properly', () => {
            const output = shallow(<AssignmentsContainer {...nextProps} />).dive();
            expect(output).toMatchSnapshot();
        });
    });

    describe('Assignments render with data', () => {
        const nextProps = {
            ...props,
            assignments: [
                {
                    assignment: {
                        score: 100,
                    },
                },
            ],
        };

        it('renders properly', () => {
            const output = shallow(<AssignmentsContainer {...nextProps} />).dive();
            expect(output).toMatchSnapshot();
        });
    });
});
