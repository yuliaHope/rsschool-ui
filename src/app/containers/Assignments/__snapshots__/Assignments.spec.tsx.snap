// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assignments container Assignments preloader renders properly 1`] = `
<div
  className="tasks"
>
  <h2>
    TASKS
  </h2>
  <Row
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="6"
    >
      <p>
        Your github private repository
        <Badge
          className="badge"
          color="dark"
          href="#"
          pill={false}
          tag="span"
        >
           
          here
        </Badge>
      </p>
    </Col>
    <Col
      className="text-right"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="6"
    >
      <p>
        You are in the TOP 50 students!
      </p>
      <p>
        Full Score: 
        0
      </p>
    </Col>
  </Row>
  <h3>
    Loading...
  </h3>
</div>
`;

exports[`Assignments container Assignments render with data renders properly 1`] = `
<div
  className="tasks"
>
  <h2>
    TASKS
  </h2>
  <Row
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="6"
    >
      <p>
        Your github private repository
        <Badge
          className="badge"
          color="dark"
          href="#"
          pill={false}
          tag="span"
        >
           
          here
        </Badge>
      </p>
    </Col>
    <Col
      className="text-right"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="6"
    >
      <p>
        You are in the TOP 50 students!
      </p>
      <p>
        Full Score: 
        100
      </p>
    </Col>
  </Row>
  <Assignments
    assignments={
      Array [
        Object {
          "assignment": Object {
            "score": 100,
          },
        },
      ]
    }
    courseId="courseId"
    updateAssignment={[MockFunction]}
  />
</div>
`;

exports[`Assignments container Initial render renders properly 1`] = `
<div
  className="tasks"
>
  <h2>
    TASKS
  </h2>
  <Row
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="6"
    >
      <p>
        Your github private repository
        <Badge
          className="badge"
          color="dark"
          href="#"
          pill={false}
          tag="span"
        >
           
          here
        </Badge>
      </p>
    </Col>
    <Col
      className="text-right"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="6"
    >
      <p>
        You are in the TOP 50 students!
      </p>
      <p>
        Full Score: 
        0
      </p>
    </Col>
  </Row>
  <Assignments
    assignments={Array []}
    courseId="courseId"
    updateAssignment={[MockFunction]}
  />
</div>
`;
